src/vs/workbench/contrib/search/browser/searchView.ts

```diff
@@ -722,8 +722,42 @@ export class SearchView extends ViewPane {
 		return this.refreshTree(event);
 	}
 
+	private reselectAfterRefreshTree(beforePreviousSelection: RenderableMatch | null): void {
+		// Selection exists -> no need to reselect as what caused the tree to refresh didn't change the selected match
+		if (this.tree.selectionSize > 0 || !beforePreviousSelection) {
+			return;
+		}
+
+		// Go to the first element of the tree
+		const iterator = this.tree.navigate();
+		let current = iterator.first();
+		// Look for the element before the previous selection in the tree
+		while (!!current) {
+			if (current.id() === beforePreviousSelection.id()) {
+				this.tree.setSelection([current]);
+				this.tree.setFocus([current]);
+				this.tree.reveal(current);
+			}
+			current = iterator.next();
+		}
+	}
+
 	refreshTree(event?: IChangeEvent): void {
 		const collapseResults = this.searchConfig.collapseResults;
+
+		// Get the selection before the refresh
+		const [previous] = this.tree.getSelection();
+		const navigator = this.tree.navigate(previous);
+		let beforePrevious = navigator.previous();
+		// Expand until beforePrevious is a Match
+		while (beforePrevious && !(beforePrevious instanceof Match)) {
+			if (this.tree.isCollapsed(beforePrevious)) {
+				this.tree.expand(beforePrevious);
+			}
```

Comment:
Not sure if this logic is correct:

[![Image from Gyazo](https://i.gyazo.com/f9e3dae5868df174ce4b59aaaff6b523.gif)](https://gyazo.com/f9e3dae5868df174ce4b59aaaff6b523)

When you expand an element, you might want to go previous from what was after it in order to get the true previous match.

Reply:
<img width="339" alt="image" src="https://github.com/user-attachments/assets/5f8168ea-e586-4691-9b77-707786c4009b">
Might also be good to test in the tree view for search.